classes/abstracts/ActionScheduler_Abstract_QueueRunner.php

```diff
@@ -68,6 +68,9 @@ public function process_action( $action_id, $context = '' ) {
 		} catch ( Exception $e ) {
 			if ( $valid_action ) {
 				$this->store->mark_failure( $action_id );
+				if ( isset( $action ) ) {
+					$this->retry_action( $action );
```

Comment:
what do you think about setting `store->mark_failure` after the retry action. so in case, we are able to schedule a retry then the action isn't failed yet. should we have another tab for action in retry? (or some other way to indicate that while the action was failed, we are retrying)